storage/innobase/log/log0recv.cc

```diff
@@ -3423,6 +3423,43 @@ void recv_reset_buffer() {
   recv_sys->recovered_offset = 0;
 }
 
+#ifdef XTRABACKUP
+static bool xb_double_bp(ulint *max_mem) {
+  if (!xtrabackup_use_free_memory_pct_set) return false;
+
+  /* find availabe free memory */
+  auto free_memory = xtrabackup::utils::free_memory();
+  xb::info() << "Maximum memory for buffer pool:" << free_memory;
+  xb::info() << "Current Buffer Pool Size: " << srv_buf_pool_size;
+
+  /* gradually increase the buffer pool until it can processed redo log in
+one iteration or it hit threshold */
+  if (srv_buf_pool_size * 2 <=
+      (free_memory * xtrabackup_use_free_memory_pct / 100)) {
+    if (dict_sys == nullptr) {
+      if (!dict_boot()) return false;
+    }
+    srv_buf_pool_size *= 2;
+
+    srv_buf_pool_size = buf_pool_size_align(srv_buf_pool_size);
+
+    xb::info() << "Increasing the buffer pool to " << srv_buf_pool_size;
+
+    buf_pool_resize();
+    recv_n_pool_free_frames *= 2;
+    *max_mem =
+        UNIV_PAGE_SIZE * (srv_buf_pool_size / UNIV_PAGE_SIZE -
+                          (recv_n_pool_free_frames * srv_buf_pool_instances));
```

Comment:
does this mean, as we double BP, more memory from BP is allocated for recovery pages and not for redo data? :thinking: